test/e2e/page-objects/pages/send-token-page.ts

```diff
@@ -0,0 +1,119 @@
+import { Driver } from '../../webdriver/driver';
+
+class SendTokenPage {
+  private driver: Driver;
+
+  private inputRecipient: string;
+
+  private inputAmount: string;
+
+  private scanButton: string;
+
+  private continueButton: object;
+
+  private ensResolvedName: string;
+
+  private ensAddressAsRecipient: string;
+
+  private ensResolvedAddress: string;
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+    this.inputAmount = '[data-testid="currency-input"]';
+    this.inputRecipient = '[data-testid="ens-input"]';
+    this.scanButton = '[data-testid="ens-qr-scan-button"]';
+    this.ensResolvedName =
+      '[data-testid="multichain-send-page__recipient__item__title"]';
+    this.ensResolvedAddress =
+      '[data-testid="multichain-send-page__recipient__item__subtitle"]';
+    this.ensAddressAsRecipient = '[data-testid="ens-input-selected"]';
+    this.continueButton = {
+      text: 'Continue',
+      tag: 'button',
+    };
+  }
+
+  async check_pageIsLoaded(): Promise<void> {
+    try {
+      await this.driver.waitForSelector(this.inputRecipient);
+      await this.driver.waitForSelector(this.scanButton);
+    } catch (e) {
+      console.log(
+        'Timeout while waiting for send token screen to be loaded',
+        e,
+      );
+      throw e;
+    }
+    console.log('Send token screen is loaded');
+  }
+
+  async fillRecipient(recipientAddress: string): Promise<void> {
+    console.log(
+      `Fill recipient input with ${recipientAddress} on send token screen`,
+    );
+    await this.driver.pasteIntoField(this.inputRecipient, recipientAddress);
+  }
+
+  async fillAmount(amount: string): Promise<void> {
+    console.log(`Fill amount input with ${amount} on send token screen`);
+    await this.driver.waitForSelector(this.inputAmount);
+    await this.driver.pasteIntoField(this.inputAmount, amount);
```

Comment:
Suggestion: can we add validate the input after paste into the field before submitting? do you think this would be valuable?

Reply:
Great catch, i have added the check in the latest commit, please check: https://github.com/MetaMask/metamask-extension/pull/25373/commits/c27938a2cb54cf5abcad98aab8e1b414d6ca6c3f